src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -157,6 +162,13 @@ public NadelExecutionStrategy(List<Service> services,
 
                 ExecutionContext newExecutionContext = buildServiceVariableOverrides(executionContext, queryTransform.getVariableValues());
 
+                String topLevelFieldId = esi.getFieldDefinition().getDefinition().getAdditionalData().get("id");
```

Comment:
we have static util method to get id I think

Reply:
we do have this method to get an id from a `Node` https://github.com/graphql-java/nadel/blob/master/src/main/java/graphql/nadel/dsl/NodeId.java which won't work here as I am getting the id from `FieldDefinition`. If I am to add something like 

```java    
    public static String getId(GraphQLFieldDefinition graphQLFieldDefinition) {
        return assertNotNull(
                graphQLFieldDefinition.getDefinition().getAdditionalData().get(ID),
                () -> String.format("expected field %s to have an id", graphQLFieldDefinition.getName()));
    }
```
some tests start breaking due to the `assertNotNull` as some mocks are not set to have an id on a definition I think. Investigating